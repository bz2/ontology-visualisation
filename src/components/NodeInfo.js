import { useState } from 'react'
import { connect } from 'redux-zero/react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import {
  BsChevronLeft,
  BsChevronRight,
} from 'react-icons/bs'
import {
  ImCross
} from 'react-icons/im'
import actions from '../store/actions'
import NodeInfoDetails from './NodeInfoDetails'

const NodeInfo = ({
  selectedNodes,
  availableNodesNormalised,
  removeFromArray
}) => {
  const { t } = useTranslation()

  const [view, setView] = useState('')
  const [nodeId, setNodeId] = useState('')

  return (
    <div className="node-info">
      <div className="node-info-navbar">
        <div className="node-info-navbar-button">
          {
            view !== '' && (
              <button
                type="button"
                title={t('goBack')}
                onClick={() => setView('')}
              >
                <BsChevronLeft />
              </button>
            )
          }
        </div>
        {
          view !== ''
            ? availableNodesNormalised[nodeId].label
            : t('selectedNodes')
        }
      </div>

      <div className="node-info-body">
        {
          view === ''
            ? selectedNodes.map((selectedNode) => {
              const selectedNodeInfo = availableNodesNormalised[selectedNode]

              if (!selectedNodeInfo) return null

              const { label, id } = selectedNodeInfo

              return (
                <div
                  className="node-info-body-row"
                  key={`selected-node-row-${id}`}
                >
                  <button
                    type="button"
                    title={t('removeSelectedNode')}
                    onClick={() => removeFromArray('selectedNodes', id)}
                  >
                    <ImCross />
                  </button>
                  <span>
                    {label}
                  </span>
                  <button
                    type="button"
                    title={t('viewNode')}
                    onClick={() => {
                      setNodeId(id)
                      setView('node')
                    }}
                  >
                    <BsChevronRight />
                  </button>
                </div>
              )
            })
            : (
              <NodeInfoDetails
                nodeId={nodeId}
              />
            )
        }
      </div>
    </div>
  )
}

NodeInfo.propTypes = {
  selectedNodes: PropTypes.arrayOf(PropTypes.string).isRequired,
  availableNodesNormalised: PropTypes.shape().isRequired,
  removeFromArray: PropTypes.func.isRequired,
}

const mapToProps = ({
  selectedNodes,
  availableNodesNormalised,
  deletedNodes
}) => ({
  selectedNodes,
  availableNodesNormalised,
  deletedNodes
})

export default connect(
  mapToProps,
  actions
)(NodeInfo)
