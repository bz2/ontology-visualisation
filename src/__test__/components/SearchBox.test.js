import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NodeInfo from '../../components/NodeInfo'

const setup = () => {
  const props = {
    searchFilter: 'abc',
    setStoreState: jest.fn()
  }

  const component = shallow(<NodeInfo {...props} />)

  return {
    component,
    props
  }
}

describe('NodeInfo', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot ', () => {
    const {
      component
    } = setup()

    expect(toJson(component)).toMatchSnapshot()
  })
})
