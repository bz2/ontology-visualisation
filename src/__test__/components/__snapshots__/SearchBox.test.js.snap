// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeInfo should match snapshot  1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "removeFromArray": [Function],
      "setStoreState": [Function],
    }
  }
  mapToProps={[Function]}
  searchFilter="abc"
  setStoreState={[MockFunction]}
>
  <Component />
</Connect>
`;
